doc/nrf/device_guides/working_with_nrf/nrf70/developing/fw_patches_ext_flash.rst

```diff
@@ -0,0 +1,167 @@
+* then it is uploaded to the nRF70 device.
+
+This feature can be enabled using DTS or :ref::`snippets` feature or by using :ref::`partition_manager`.
+
+Configuration
+-------------
+
+The following configuration options are available:
+
+* :kconfig:option:`CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE`: Enables the option to store the firmware patch in external non-XIP memory.
+* :kconfig:option:`CONFIG_NRF_WIFI_FW_FLASH_CHUNK_SIZE`: Defines the size of the chunks used to read the firmware patches from the external non-XIP memory.
+  The default value is 8192 bytes.
+
+One of the following options must be enabled:
+
+* ``nrf70_fw_partition`` : Defines the name of the external memory partition that stores the firmware patches.
+  This must be defined in the devicetree.
+
+* ``nrf70_wifi_fw`` : Defines the name of the external memory partition that stores the firmware patches.
+  This must be defined in the partition manager configuration file.
+
+Building
+--------
+
+See :ref:`nrf7002dk_nrf5340` for general instructions on building.
+
+Additionally, you must enable either the `nrf70-fw-patch-ext-flash` snippet or the :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` option.
+
+For example, to build the :ref:`wifi_shell_sample` sample for the nRF5340 DK with the ``nrf70-fw-patch-ext-flash`` snippet enabled, run the following commands:
+
+With west
+^^^^^^^^^
+
+.. code-block:: console
+
+    west build -p -b nrf5340dk_nrf5340_cpuapp -S nrf70-fw-patch-ext-flash samples/wifi/shell -- -DSHIELD=nrf7002ek
+
+With CMake
+^^^^^^^^^^
+
+.. code-block:: console
+
+    cmake -GNinja -Bbuild -DBOARD=nrf5340dk_nrf5340_cpuapp -DSHIELD=nrf7002ek -DSNIPPET=nrf70-fw-patch-ext-flash samples/wifi/shell
+    ninja -C build
+
+For example, to build the :ref:`wifi_shell_sample` sample for the nRF5340 DK with the :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` option enabled, run the following commands:
+
+With west
+^^^^^^^^^
+
+.. code-block:: console
+
+    west build -p -b nrf5340dk_nrf5340_cpuapp samples/wifi/shell -- -DSHIELD=nrf7002ek -DCONFIG_PARTITION_MANAGER_ENABLED=y -DCONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE=y
+
+With CMake
+^^^^^^^^^^
+
+.. code-block:: console
+
+    cmake -GNinja -Bbuild -DBOARD=nrf5340dk_nrf5340_cpuapp -DSHIELD=nrf7002ek -DCONFIG_PARTITION_MANAGER_ENABLED=y -DCONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE=y samples/wifi/shell
+    ninja -C build
+
+Programming
+-----------
+
+To program the firmware image with the firmware patches stored in the external memory, use the following commands:
+
+With west
+^^^^^^^^^
+
+No changes are needed to the programming command:
+
+.. code-block:: console
+
+    west flash
+
+With other tools
+^^^^^^^^^^^^^^^^
+
+You must use the :file:`merged.hex` file instead of the :file:`zephyr.hex` file to choose the program image explicitly.
+
+For example, for nrfjprog:
+
+.. code-block:: console
+
+    nrfjprog -f nrf53 -s 0 --program build/zephyr/merged.hex --chiperase --reset
```

Comment:
This will do nothing to QSPI, so if there is existing data there, you will just end up with corrupt data. You could do a full QSPI erase but who wants to sit around for 2 minutes? So add `--qspisectorerase`